diff --git a/language/src/module/e_mod_lang.c b/language/src/module/e_mod_lang.c
index a3ec833..30724ba 100644
--- a/language/src/module/e_mod_lang.c
+++ b/language/src/module/e_mod_lang.c
@@ -14,7 +14,7 @@
 #ifdef __FreeBSD__
    #define DFLT_XKB_RULES_FILE   "/usr/X11R6/lib/X11/xkb/rules/xfree86"
 #else
-   #define DFLT_XKB_RULES_FILE   "/etc/X11/xkb/rules/xfree86"
+   #define DFLT_XKB_RULES_FILE   "/usr/share/X11/xkb/rules/xfree86"
 #endif // __FreeBSD__
 #define DFLT_XKB_LAYOUT	      "us"
 #define DFLT_XKB_MODEL	      "pc101"
@@ -206,7 +206,7 @@ lang_load_xfree_language_kbd_layouts(Config *cfg)
    if (!exml) return;
    if (!exml_init(exml)) EXML_RETURN_ON_ERROR(exml);
 
-   result = exml_file_read(exml, "/etc/X11/xkb/rules/xfree86.xml");
+   result = exml_file_read(exml, "/usr/share/X11/xkb/rules/xfree86.xml");
    if (!result || result == -1)
      { 
 #ifdef __FreeBSD__
@@ -296,7 +296,7 @@ lang_load_xfree_kbd_models(Config *cfg)
    if (!exml_init(exml))
      EXML_RETURN_ON_ERROR(exml);
 
-   result = exml_file_read(exml, "/etc/X11/xkb/rules/xfree86.xml");
+   result = exml_file_read(exml, "/usr/share/X11/xkb/rules/xfree86.xml");
    if (!result || result == -1)
      { 
 #ifdef __FreeBSD__
