diff --git a/config/illume/e.src b/config/illume/e.src
index f0d577a..e1073d4 100644
--- a/config/illume/e.src
+++ b/config/illume/e.src
@@ -257,7 +257,7 @@ group "E_Config" struct {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
       value "modifiers" int: 0;
-      value "key" string: "Execute";
+      value "key" string: "XF86PowerOff";
       value "action" string: "delayed_action";
       value "params" string: "[0.0 syscon] [2.0 halt_now]";
       value "any_mod" uchar: 1;
