Index: terminology/data/themes/default.edc
===================================================================
--- terminology/data/themes/default.edc	(revision 72971)
+++ terminology/data/themes/default.edc	(working copy)
@@ -41,7 +41,7 @@
          part { name: "base"; type: RECT;
             mouse_events: 1;
             description { state: "default" 0.0;
-               color: 48 48 48 255;
+               color: 5 5 5 255;
             }
             description { state: "translucent" 0.0;
                inherit: "default" 0.0;
@@ -214,7 +214,7 @@
                min: 32 32;
                max: 32 32;
                rel2.offset: -1 -1;
-               align: 1.0 1.0;
+               align: 1.0 0.0;
                image.normal: "bg_led_base.png";
             }
             description { state: "visible" 0.0;
