From ff801b77b7d246ce6b0db93fcdb74cd6cb68667c Mon Sep 17 00:00:00 2001
From: Albin Tonnerre <albin.tonnerre@gmail.com>
Date: Fri, 19 Jun 2009 11:13:44 +0200
Subject: [PATCH 1/2] fix library soname

---
 configure.ac |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 2e6fca4..66a177f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -10,6 +10,7 @@ AC_CANONICAL_HOST
 AC_ISC_POSIX
 
 AM_INIT_AUTOMAKE(1.6 dist-bzip2)
+AM_MAINTAINER_MODE([enable])
 AM_CONFIG_HEADER(config.h)
 
 AC_LIBTOOL_WIN32_DLL
@@ -28,7 +29,7 @@ AC_SUBST(version_info)
 
 ### Needed information
 
-release="ver-svn-02"
+release="ver-pre-01"
 case "$host_os" in
    mingw32ce* | cegcc*)
       MODULE_ARCH="$host_os-$host_cpu"
-- 
1.6.3.1

