diff --git a/src/lib/efreet_icon.c b/src/lib/efreet_icon.c
index af8fdfa..177d073 100644
--- a/src/lib/efreet_icon.c
+++ b/src/lib/efreet_icon.c
@@ -119,7 +119,7 @@ efreet_icon_init(void)
 
     if (!efreet_icon_themes)
     {
-        const char *default_exts[] = {".png", ".xpm", NULL};
+        const char *default_exts[] = {".png", ".svg", ".xpm", NULL};
         int i;
 
         if (!ecore_init())
