From b4045c49f8a97561d06306f2e5a52f239f8f992f Mon Sep 17 00:00:00 2001
From: Carsten Haitzler <raster@openmoko.org>
Date: Fri, 10 Oct 2008 00:38:23 +1100
Subject: [PATCH] disable program cache - really not a good idea i think.

---
 packages/efl1/edje_svn.bb |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/packages/efl1/edje_svn.bb b/packages/efl1/edje_svn.bb
index 664e965..e429cf7 100644
--- a/packages/efl1/edje_svn.bb
+++ b/packages/efl1/edje_svn.bb
@@ -2,12 +2,13 @@ DESCRIPTION = "Edje is the Enlightenment graphical design & layout library"
 DEPENDS = "eet evas ecore embryo edje-native"
 LICENSE = "MIT BSD"
 PV = "0.5.0.043+svnr${SRCREV}"
-PR = "r4"
+PR = "r6"
 
 inherit efl
 
-# gain some extra performance at the expense of RAM
-EXTRA_OECONF = "--enable-edje-program-cache"
+# gain some extra performance at the expense of RAM - generally i'd say bad
+# and a possible source of bugs
+#EXTRA_OECONF = "--enable-edje-program-cache"
 
 PACKAGES =+ "${PN}-utils"
 DEBIAN_NOAUTONAME_${PN}-utils = "1"
-- 
1.5.4.3

