From 29d4f425b2250ed3741fffb43a3f631594fe7673 Mon Sep 17 00:00:00 2001
From: Carsten Haitzler <raster@openmoko.org>
Date: Tue, 14 Apr 2009 20:31:03 +1000
Subject: [PATCH] xserver-kdrive-common: fix tscalibrate script to not run if ts dev not exist

---
 .../xserver-kdrive-common/30xTs_Calibrate          |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/recipes/xserver-kdrive-common/xserver-kdrive-common/30xTs_Calibrate b/recipes/xserver-kdrive-common/xserver-kdrive-common/30xTs_Calibrate
index 8e6a8b2..7078690 100644
--- a/recipes/xserver-kdrive-common/xserver-kdrive-common/30xTs_Calibrate
+++ b/recipes/xserver-kdrive-common/xserver-kdrive-common/30xTs_Calibrate
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-while [ ! -z $TSLIB_TSDEVICE ] && [ ! -f /etc/pointercal ]
+while [ ! -z $TSLIB_TSDEVICE ] && [ -f $TSLIB_TSDEVICE] && [ ! -f /etc/pointercal ]
 do
    /usr/bin/xtscal
 done
-- 
1.5.6.3

