From 36e3b6aa6885b53023c81e66a1443625c3aba7be Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster) <raster@rasterman.com>
Date: Mon, 1 Dec 2008 10:42:48 +1100
Subject: [PATCH] e-wm: fix package version to match src

---
 packages/e17/e-wm_svn.bb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/packages/e17/e-wm_svn.bb b/packages/e17/e-wm_svn.bb
index f723f90..ee9de91 100644
--- a/packages/e17/e-wm_svn.bb
+++ b/packages/e17/e-wm_svn.bb
@@ -1,8 +1,8 @@
 DESCRIPTION = "The Enlightenment Window Manager Version 17"
 DEPENDS = "eet evas ecore edje efreet edbus"
 LICENSE = "MIT BSD"
-PV = "0.16.999.043+svnr${SRCREV}"
-PR = "r22"
+PV = "0.16.999.050+svnr${SRCREV}"
+PR = "r1"
 
 inherit e update-alternatives
 
-- 
1.5.4.3

