From e4699630a91a3f0c3e1510173faff254ca24843e Mon Sep 17 00:00:00 2001
From: Carsten Haitzler <raster@openmoko.org>
Date: Fri, 10 Oct 2008 14:47:53 +1100
Subject: [PATCH] revert expedite change...

---
 packages/e17/expedite_svn.bb |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git a/packages/e17/expedite_svn.bb b/packages/e17/expedite_svn.bb
index 62f3f25..1337487 100644
--- a/packages/e17/expedite_svn.bb
+++ b/packages/e17/expedite_svn.bb
@@ -7,9 +7,6 @@ PR = "r0"
 
 inherit e
 
-SRC_URI = "svn://svn.enlightenment.org/svn/e/trunk;module=expedite;proto=http"
-S = "${WORKDIR}/expedite"
-	
 EXTRA_OECONF = "\
   --x-includes=${STAGING_INCDIR}/X11 \
   --x-libraries=${STAGING_LIBDIR} \
-- 
1.5.4.3

