From 02683a44f10416f772838bf6a62081fb738505d8 Mon Sep 17 00:00:00 2001
From: Carsten Haitzler <raster@openmoko.org>
Date: Mon, 29 Sep 2008 23:42:26 +1000
Subject: [PATCH] fix ttf font package build. works now.

---
 packages/ttf-fonts/ttf.inc |   12 ++++++++----
 1 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/packages/ttf-fonts/ttf.inc b/packages/ttf-fonts/ttf.inc
index e2be378..f4c3f24 100644
--- a/packages/ttf-fonts/ttf.inc
+++ b/packages/ttf-fonts/ttf.inc
@@ -4,10 +4,14 @@ RRECOMMENDS += "font-update-common"
 
 do_install() {
     install -d ${D}${datadir}/fonts/truetype/
-    for f in *.ttf *.ttc
-       do
-          [ -f $f ] && install -m 0644 $f ${D}${datadir}/fonts/truetype/
-       done
+    FILES=`ls *.ttf`
+    if [ -n "$FILES" ]; then
+        install -m 0644 $FILES ${D}${datadir}/fonts/truetype/
+    fi
+    FILES=`ls *.ttc || true`
+    if [ -n "$FILES" ]; then
+        install -m 0644 $FILES ${D}${datadir}/fonts/truetype/
+    fi
 }
 
 pkg_postinst_append() {
-- 
1.5.4.3

