From 5e38d6030e5dc24992445d638aa6ccb6fa03acf2 Mon Sep 17 00:00:00 2001
From: Carsten Haitzler <raster@openmoko.org>
Date: Tue, 30 Sep 2008 22:45:57 +1000
Subject: [PATCH] disable no-cursor patch - will break e now that its upstream.

---
 packages/e17/e-wm_svn.bb |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/packages/e17/e-wm_svn.bb b/packages/e17/e-wm_svn.bb
index 46ade70..e16cad0 100644
--- a/packages/e17/e-wm_svn.bb
+++ b/packages/e17/e-wm_svn.bb
@@ -2,7 +2,7 @@ DESCRIPTION = "The Enlightenment Window Mananger Version 17"
 DEPENDS = "eet evas ecore edje efreet edbus"
 LICENSE = "MIT BSD"
 PV = "0.16.999.043+svnr${SRCREV}"
-PR = "r13"
+PR = "r14"
 
 inherit e update-alternatives
 
@@ -10,7 +10,6 @@ RDEPENDS += "shared-mime-info mime-support"
 
 SRC_URI = "\
   svn://svn.enlightenment.org/svn/e/trunk;module=e;proto=http \
-  file://disable-e-cursor.patch;patch=1 \
   file://enlightenment_start.oe \
   file://applications.menu \
 "
-- 
1.5.4.3

