From c13444d1078f46a97c5dfe719f1942671c08106f Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster) <raster@rasterman.com>
Date: Mon, 1 Dec 2008 10:47:45 +1100
Subject: [PATCH] efreet: fix package version to match src

---
 packages/efl1/efreet_svn.bb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/packages/efl1/efreet_svn.bb b/packages/efl1/efreet_svn.bb
index 37e9443..218d22e 100644
--- a/packages/efl1/efreet_svn.bb
+++ b/packages/efl1/efreet_svn.bb
@@ -1,8 +1,8 @@
 DESCRIPTION = "The Enlightenment freedesktop.org library"
 DEPENDS = "ecore"
 LICENSE = "MIT BSD"
-PV = "0.5.0.043+svnr${SRCREV}"
-PR = "r2"
+PV = "0.5.0.050+svnr${SRCREV}"
+PR = "r1"
 
 inherit efl
 
-- 
1.5.4.3

