From 42b086135e6dd387e42189a6e9994e601d473142 Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster) <raster@rasterman.com>
Date: Mon, 1 Dec 2008 10:53:49 +1100
Subject: [PATCH] embryo: fix package version to match src

---
 packages/efl1/embryo_svn.bb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/packages/efl1/embryo_svn.bb b/packages/efl1/embryo_svn.bb
index 956ff77..688ec22 100644
--- a/packages/efl1/embryo_svn.bb
+++ b/packages/efl1/embryo_svn.bb
@@ -1,7 +1,7 @@
 DESCRIPTION = "The Enlightenment C-like scripting language for Edje"
 LICENSE = "MIT BSD"
-PV = "0.9.9.043+svnr${SRCREV}"
-PR = "r2"
+PV = "0.9.9.050+svnr${SRCREV}"
+PR = "r1"
 
 inherit efl
 
-- 
1.5.4.3

