From c9db0844f8c660a6f5e466c751d43178ae36b717 Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster) <raster@rasterman.com>
Date: Mon, 1 Dec 2008 10:54:31 +1100
Subject: [PATCH] emotion: fix package version to match src

---
 packages/efl1/emotion_svn.bb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/packages/efl1/emotion_svn.bb b/packages/efl1/emotion_svn.bb
index c1d3ff3..455bc82 100644
--- a/packages/efl1/emotion_svn.bb
+++ b/packages/efl1/emotion_svn.bb
@@ -2,8 +2,8 @@ DESCRIPTION = "The Enlightenment multimedia library"
 LICENSE = "MIT BSD"
 # we no longer build the libxine backend, since the gstreamer backend seems more promising
 DEPENDS = "eet evas ecore edje gstreamer gst-plugins-base"
-PV = "0.1.0+svnr${SRCREV}"
-PR = "r3"
+PV = "0.1.0.042+svnr${SRCREV}"
+PR = "r1"
 
 inherit efl
 
-- 
1.5.4.3

