From e87c95b2b484e12db2111f23342e7bb29ac3de4f Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster) <raster@rasterman.com>
Date: Mon, 1 Dec 2008 10:55:05 +1100
Subject: [PATCH] esmart: fix package version to match src

---
 packages/efl1/esmart_svn.bb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/packages/efl1/esmart_svn.bb b/packages/efl1/esmart_svn.bb
index d0983e0..1ff75a4 100644
--- a/packages/efl1/esmart_svn.bb
+++ b/packages/efl1/esmart_svn.bb
@@ -1,8 +1,8 @@
 DESCRIPTION = "ESmart is a collection of smart Evas objects"
 LICENSE = "MIT BSD"
 DEPENDS = "evas ecore edje imlib2 epsilon libtool"
-PV = "0.9.0.043+svnr${SRCREV}"
-PR = "r3"
+PV = "0.9.0.050+svnr${SRCREV}"
+PR = "r1"
 
 inherit efl
 
-- 
1.5.4.3

