From 8975e90309bc11a85fa30f9178a8750165e61265 Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster) <raster@rasterman.com>
Date: Mon, 1 Dec 2008 10:56:48 +1100
Subject: [PATCH] ewl: fix package version to match src

---
 packages/efl1/ewl_svn.bb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/packages/efl1/ewl_svn.bb b/packages/efl1/ewl_svn.bb
index 336d3ed..6425e86 100644
--- a/packages/efl1/ewl_svn.bb
+++ b/packages/efl1/ewl_svn.bb
@@ -1,8 +1,8 @@
 DESCRIPTION = "The Enlightened Widget Library"
 DEPENDS = "evas ecore edje emotion efreet epsilon"
 LICENSE = "MIT BSD"
-PV = "0.5.2.042+svnr${SRCREV}"
-PR = "r3"
+PV = "0.5.2.050+svnr${SRCREV}"
+PR = "r1"
 
 inherit efl
 
-- 
1.5.4.3

