From 57df5bf74c1b3b94f9f68ae82d59a9954adeccec Mon Sep 17 00:00:00 2001
From: Carsten Haitzler <raster@openmoko.org>
Date: Wed, 29 Oct 2008 10:46:16 +1100
Subject: [PATCH] rebuild initscripts with ldconfig etc. removed.

---
 packages/initscripts/initscripts_1.0.bb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/packages/initscripts/initscripts_1.0.bb b/packages/initscripts/initscripts_1.0.bb
index 405cd46..eb393b8 100644
--- a/packages/initscripts/initscripts_1.0.bb
+++ b/packages/initscripts/initscripts_1.0.bb
@@ -4,7 +4,7 @@ PRIORITY = "required"
 DEPENDS = "makedevs"
 RDEPENDS = "makedevs"
 LICENSE = "GPL"
-FILE_PR = "r109"
+FILE_PR = "r110"
 
 SRC_URI = "file://functions \
            file://halt \
-- 
1.5.4.3

