From 3fe6158d1888aa57643acdc18b6ffba3aaaf88d7 Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster) <raster@rasterman.com>
Date: Thu, 4 Dec 2008 12:55:47 +1100
Subject: [PATCH] e-wm: sysactions package is "all" archs, not machne-specific and preload.a is not used for anything at all. not even dev.

---
 packages/e17/e-wm_svn.bb |    7 +++----
 1 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/packages/e17/e-wm_svn.bb b/packages/e17/e-wm_svn.bb
index ee9de91..4902f6c 100644
--- a/packages/e17/e-wm_svn.bb
+++ b/packages/e17/e-wm_svn.bb
@@ -2,7 +2,7 @@ DESCRIPTION = "The Enlightenment Window Manager Version 17"
 DEPENDS = "eet evas ecore edje efreet edbus"
 LICENSE = "MIT BSD"
 PV = "0.16.999.050+svnr${SRCREV}"
-PR = "r1"
+PR = "r4"
 
 inherit e update-alternatives
 
@@ -39,6 +39,7 @@ RRECOMMENDS_${PN} = "\
   ${PN}-icons \
   ${PN}-other \
   ${PN}-input-methods \
+  ${PN}-sysactions \
 "
 
 PACKAGE_ARCH_${PN}-config-default = "all"
@@ -55,7 +56,7 @@ PACKAGE_ARCH_${PN}-images = "all"
 PACKAGE_ARCH_${PN}-icons = "all"
 PACKAGE_ARCH_${PN}-other = "all"
 PACKAGE_ARCH_${PN}-input-methods = "all"
-PACKAGE_ARCH_${PN}-sysactions = "${MACHINE_ARCH}"
+PACKAGE_ARCH_${PN}-sysactions = "all"
 
 SRC_URI = "\
   svn://svn.enlightenment.org/svn/e/trunk;module=e;proto=http \
@@ -136,8 +137,6 @@ FILES_${PN}-dbg += "\
   ${libdir}/enlightenment/modules/*/*/.debug/ \
   ${libdir}/enlightenment/preload/.debug/ \
 "
-FILES_${PN}-dev += "${libdir}/enlightenment/preload/*.?a"
-
 CONFFILES_${PN} = "/etc/xdg/menus/applications.menu"
 
 ALTERNATIVE_PATH = "${bindir}/enlightenment_start.oe"
-- 
1.5.4.3

