From c471ed659edb7913c5edec6afd32871ee0fcb9ea Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster <raster@rasterman.com>
Date: Mon, 26 Jan 2009 10:14:31 +1100
Subject: [PATCH] linux-omap-2.6.28: disable as usbhost is broken pretty badly.

---
 packages/linux/linux-omap_2.6.28.bb |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/packages/linux/linux-omap_2.6.28.bb b/packages/linux/linux-omap_2.6.28.bb
index 0924772..6051669 100644
--- a/packages/linux/linux-omap_2.6.28.bb
+++ b/packages/linux/linux-omap_2.6.28.bb
@@ -8,6 +8,7 @@ COMPATIBLE_MACHINE = "omap5912osk|omap1710h3|omap2430sdp|omap2420h4|beagleboard|
 DEFAULT_PREFERENCE = "-1"
 DEFAULT_PREFERENCE_beagleboard = "1"
 
+
 SRCREV = "79d042a081d3e467c735bb0d9569ed6296f85a3c"
 
 PV = "2.6.28"
-- 
1.5.4.3

