From 131cc458ef3fa5641787a8a6b98bec7d7ed78502 Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster) <raster@rasterman.com>
Date: Mon, 1 Dec 2008 09:17:08 +1100
Subject: [PATCH] illume-image: add swisstelecom libefso + dialler.

---
 packages/images/illume-image.bb |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/packages/images/illume-image.bb b/packages/images/illume-image.bb
index fbb14d8..a189ece 100644
--- a/packages/images/illume-image.bb
+++ b/packages/images/illume-image.bb
@@ -137,10 +137,10 @@ PYTHON_INSTALL = "\
 
 # zhone
 ZHONE_INSTALL = "\
-#  gsm0710muxd \
-#  frameworkd \
-#  fso-gpsd \
-##  zhone \
+  gsm0710muxd \
+  frameworkd \
+  fso-gpsd \
+#  zhone \
 "
 
 # additional apps
@@ -156,6 +156,8 @@ APPS_INSTALL = "\
    elementary-alarm \
    gpe-scap \
    openmoko-calculator2 \
+   libefso \
+   essential-dialer \
 "
 
 IMAGE_INSTALL = "\
-- 
1.5.4.3

