From a7bd061046b3ac69b7993b3c07ab36103c5dfdb6 Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster <raster@rasterman.com>
Date: Mon, 1 Dec 2008 19:19:51 +1100
Subject: [PATCH] qemu: fix qemu to build again. otherwise qemu fails with assembler errors

---
 packages/qemu/qemu_svn.bb |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/packages/qemu/qemu_svn.bb b/packages/qemu/qemu_svn.bb
index b4acae7..fa06f59 100644
--- a/packages/qemu/qemu_svn.bb
+++ b/packages/qemu/qemu_svn.bb
@@ -3,6 +3,8 @@ DEPENDS = "zlib"
 PV = "0.9.1+svnr${SRCREV}"
 PR = "r14"
 
+DEFAULT_PREFERENCE = "-99"
+
 FILESPATH = "${FILE_DIRNAME}/qemu-${PV}/:${FILE_DIRNAME}/qemu-0.9.1+svn/"
 
 SRC_URI = "\
-- 
1.5.6.3

