From 7b8b1221a01b5ad4569b134a49db267b9e9c2dbb Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster <raster@rasterman.com>
Date: Thu, 4 Dec 2008 12:54:38 +1100
Subject: [PATCH] illume-image: clean up language setup for translations and remove unsued pkgs

---
 packages/images/illume-image.bb |   26 ++++++--------------------
 1 files changed, 6 insertions(+), 20 deletions(-)

diff --git a/packages/images/illume-image.bb b/packages/images/illume-image.bb
index 857d5b0..549fb78 100644
--- a/packages/images/illume-image.bb
+++ b/packages/images/illume-image.bb
@@ -17,10 +17,6 @@ BASE_INSTALL = "\
   exquisite \
   exquisite-themes \
   exquisite-theme-illume \
-#  rsync \
-#  screen \
-#  fbset \
-#  fbset-modes \
 "
 
 # Some machines don't set a *runtime* provider for X, so default to Xfbdev here
@@ -30,7 +26,10 @@ XSERVER ?= "xserver-kdrive-fbdev"
 # getting an X window system up
 X_INSTALL = "\
   glibc-charmap-utf-8 \
+  shared-mime-info \
+  mime-support \
   e-wm \
+  e-wm-sysactions \
   e-wm-config-illume \
   e-wm-config-standard \
   e-wm-config-netbook \
@@ -57,18 +56,14 @@ X_INSTALL = "\
 
 # useful command line tools
 TOOLS_INSTALL = "\
-#  bash \
   dosfstools \
-#  iptables \
   lsof \
   mickeydbus \
-  mickeyterm \
   mtd-utils \
   nano \
   powertop \
   s3c24xx-gpio \
   sysstat \
-#  tcpdump \
 "
 
 # audio
@@ -116,23 +111,15 @@ ZHONE_INSTALL = "\
   fso-gsm0710muxd \
   frameworkd \
   fso-gpsd \
-#  zhone \
+  zhone \
 "
 
 # additional apps
 APPS_INSTALL = "\
-#  tichy \
-#  gpe-gallery \
-#  gpe-sketchbook \
-#  gpe-filemanager \
-#  vagalume \
-#  starling \
-   rxvt-unicode \
-   gpe-terminal \
    elementary-alarm \
    gpe-scap \
-   libefso \
-   essential-dialer \
+#   libefso \
+#   essential-dialer \
 "
 
 IMAGE_INSTALL = "\
@@ -156,7 +143,6 @@ fso_rootfs_postprocess() {
     date "+%m%d%H%M%Y" >./etc/timestamp
     # alias foo
     echo "alias pico=nano" >>./etc/profile
-#    echo "alias fso='cd /local/pkg/fso'" >>./etc/profile
     echo "alias ipkg='opkg'" >>./etc/profile
     # dns
     echo "nameserver 208.67.222.222" >>./etc/resolv.conf
-- 
1.5.6.3

