From 1d3759e3e59e7536dc0e7a6feefc357043c79075 Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster <raster@rasterman.com>
Date: Thu, 4 Dec 2008 12:55:47 +1100
Subject: [PATCH] e-wm: sysactions package is "all" archs, not machne-specific and preload.a is not used for anything at all. not even dev.

---
 packages/e17/e-wm_svn.bb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/packages/e17/e-wm_svn.bb b/packages/e17/e-wm_svn.bb
index e82cf69..fa6d042 100644
--- a/packages/e17/e-wm_svn.bb
+++ b/packages/e17/e-wm_svn.bb
@@ -2,7 +2,7 @@ DESCRIPTION = "The Enlightenment Window Manager Version 17"
 DEPENDS = "eet evas ecore edje efreet edbus"
 LICENSE = "MIT BSD"
 PV = "0.16.999.050+svnr${SRCREV}"
-PR = "r3"
+PR = "r4"
 
 inherit e update-alternatives
 
-- 
1.5.6.3

