From 7a150c4a8ed47a5011d1e0c53988a2d60c47bb1b Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster <raster@rasterman.com>
Date: Thu, 19 Feb 2009 21:56:54 +1100
Subject: [PATCH] elementary: explicitly refer to the edje_cc build in staging

---
 packages/efl1/elementary_svn.bb |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/packages/efl1/elementary_svn.bb b/packages/efl1/elementary_svn.bb
index eb880ae..0597ffc 100644
--- a/packages/efl1/elementary_svn.bb
+++ b/packages/efl1/elementary_svn.bb
@@ -30,3 +30,7 @@ FILES_${PN}-tests = "\
   ${datadir}/applications/* \
   ${datadir}/icons/* \
 "
+
+EXTRA_OECONF = "\
+  --with-edje-cc=${STAGING_BINDIR_NATIVE}/edje_cc \
+"
-- 
1.5.6.3

