diff --git a/configure.ac b/configure.ac
index c16c205..77b957c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -10,6 +10,7 @@ AC_CANONICAL_HOST
 AC_ISC_POSIX
 
 AM_INIT_AUTOMAKE(1.6 dist-bzip2)
+AM_MAINTAINER_MODE([enabled])
 AM_CONFIG_HEADER(config.h)
 
 AC_LIBTOOL_WIN32_DLL
@@ -29,7 +30,7 @@ case "$host_os" in
    mingw32ce* | cegcc*)
       ;;
    *)
-      release_info="-release $release"
+      release_info=""
       ;;
 esac
 AC_SUBST(release_info)
