diff -Naur exquisite/demo/run-demo.sh exquisite.new//demo/run-demo.sh
--- exquisite/demo/run-demo.sh	2012-03-23 07:54:12.843479052 +0000
+++ exquisite.new//demo/run-demo.sh	2012-03-23 07:57:02.828579544 +0000
@@ -1,12 +1,12 @@
 #!/bin/sh
 
-# to avoid needing an installl- lets run this from the src dir
-MAIN="../src/bin/exquisite -verbose"
-WRITE="../src/bin/exquisite-write"
+PREFIX=/usr
+MAIN="$PREFIX/bin/exquisite -verbose"
+WRITE="$PREFIX/bin/exquisite-write"
 
 # start up gui - some examples here for different resolutions and the fb
 #$MAIN -t ~/C/SVN/THEMES/b_and_w/exquisite-ang.edj -x 480 -y 640 &
-$MAIN -t ../data/themes/default.edj -x 480 -y 640 -cleanvt &
+$MAIN -t $PREFIX/share/exquisite/data/themes/default.edj -x 480 -y 640 -cleanvt &
 #$MAIN -t ../data/themes/default.edj -x 640 -y 480 &
 #$MAIN -t ../data/themes/default.edj -x 240 -y 320 &
 #$MAIN -t ../data/themes/default.edj -x 1280 -y 1024 &
